lib/ddtrace/contrib/auto_instrument.rb

```diff
@@ -0,0 +1,45 @@
+require 'ddtrace'
+
+module Datadog
+  module Contrib
+    # Extensions for auto instrumentation added to the base library
+    # AutoInstrumentation enables all integration
+    module AutoInstrument
+      def self.extended(base)
+        base.send(:extend, Patch)
+      end
+
+      # Patch adds method for invoking auto_instrumentation
+      module Patch
+        def add_auto_instrument
+          if Datadog::Contrib::Rails::Utils.railtie_supported?
+            require 'ddtrace/contrib/rails/auto_instrument_railtie'
+          else
+            AutoInstrument.patch_all
+          end
+        end
+      end
```

Comment:
I think that the way we are composing `add_auto_instrument` doesn't seem like it would work nicely if another implementation tries to add new behaviour that also happen during `add_auto_instrument`.

I suggest we call `super` (if defined) in `add_auto_instrument` to ensure that the parent implementation is also called.